src/utils/queueUtil.ts

```diff
@@ -0,0 +1,29 @@
+import type { TaskItemImpl } from '@/stores/queueStore'
+import type { JobState } from '@/types/queue'
+
+/**
+ * Map a task to a UI job state, including initialization override.
+ *
+ * @param task Task item from the queue store
+ * @param isInitializing True if the prompt is currently initializing
+ * @returns JobState for UI
+ */
+export const jobStateFromTask = (
+  task: TaskItemImpl,
+  isInitializing: boolean
+): JobState => {
+  if (isInitializing) return 'initialization'
+  const status = task.displayStatus
+  switch (status) {
+    case 'Running':
+      return 'running'
+    case 'Pending':
+      return 'queued'
```

Comment:
Question: Is it possible to standardize so that it's either always `queued` or always `pending`?